### ISSUE TITLE
(ecs): .dockerignore file using inclusion breaks asset build. (also affects ecr-assets module)

### ISSUE THREAD

**Issue by gbvanrenswoude** [aws/aws-cdk#18699]
### What is the problem?

When using ContainerAssets e.g. like 
```ts
image: ecs.ContainerImage.fromAsset(__dirname, {
                ignoreMode: IgnoreMode.DOCKER
            }),
```

and using .dockerignore file like:
```
**/*
!interfaces/**/*
!src/**/*
!test/**/*
!*.yml
!*.js
!*.json
!*.pem
```

instead of just ignoring by exclusion, the asset formed to run the docker build against during deploy time will not contain all files. It will miss folders, usually resulting in the `docker build` command ran in the asset operation to break.

### Reproduction Steps

Follow the steps posted in the issue.
cdk.json and the entrypoint to the app reside in the root of the project. The code of the application is in a src folder. The Dockerfile resides in the root of the application as well. 

### What did you expect to happen?

Bundling the container as a containerImageAsset in cdk.

### What actually happened?

Error during bundle: `error: Could not resolve "src/main.ts"`

### CDK CLI Version

2.8.0

### Framework Version

2.8.0

### Node.js Version

17

### OS

MacOs

### Language

Typescript

**Comment by RichiCoder1** [comment]
Just ran into this. Was very surprising behavior, especially since it's a default

**Comment by RichiCoder1** [comment]
Still valid I believe

**Comment by amw** [comment]
Better support for .dockerignore files is still relevant and important

**Comment by aemada-aws** [comment]
@amw The bot mentioned it as a duplicate of https://github.com/aws/aws-cdk/issues/20712 to consolidate discussions in 1 issue. I understand it is still relevant, but since the other issue describes the same problem and has more engagement (more comments), I suggest we only keep issue https://github.com/aws/aws-cdk/issues/20712.

Please correct me if I'm wrong and this issue adds more information that is not present in #20712.

**Comment by amw** [comment]
@aemada-aws You're right, it does seem to be closely related. I think we can close this one as duplicate.
